src/Passes/Source/Rules/Tests/Unit/RemoveFunctionCall.cpp

```diff
@@ -51,9 +51,10 @@ TEST(RuleSetTestSuite, RemovingFunctionCall)
         factory.removeFunctionCall("__quantum__qis__h__body");
     };
 
-    auto profile = std::make_shared<RuleSetProfile>(std::move(configure_profile));
-    ir_manip->applyProfile(profile);
+    auto profile = std::make_shared<DefaultProfileGenerator>(std::move(configure_profile));
 
+    ir_manip->applyProfile(profile);
+    llvm::errs() << *ir_manip->module() << "\n";
```

Comment:
Is there a check somewhere that there are no errors?

Reply:
Yes,
```c++
    void IrManipulationTestHelper::applyProfile(
        GeneratorPtr const&      generator,
        OptimizationLevel const& optimisation_level,
        bool                     debug)
    {
        auto profile = generator->newProfile(optimisation_level, debug);
        profile.apply(*module_);

        // Verifying that the module is valid
        if (isModuleBroken())
        {
            throw std::runtime_error("Module was broken after applying result");
        }
    }
```